### PR TITLE
feat(datasource/adoptium): support jre only releases

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

You can now pass `java-jre` as `lookupName` to only get releases with JRE available. Currently only LTS releases have a JRE available.

## Context:

We need this for our docker images. The JRE is typically only ~60MB where a JDK is about ~300MB.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/adoptium-java/readme.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 This datasource returns releases from [Adoptium](https://adoptium.net/) API.
 
-It uses `project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
+It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
 
 It only uses the first 50 pages with 50 items per page.
+
+Use `java-jdk` or `java` as `lookupName` to get releases with JDK available or use `java-jre` to only get releases with JRE available.
```
Comment (HonkingGoose) [inline lib/datasource/adoptium-java/readme.md:7]:
```suggestion
Use `java-jdk` or `java` as `lookupName` to get releases which come with the Java Development Kit.

Use `java-jre`  as `lookupName` if you only want releases which come with the Java Runtime Environment.
```

Maybe something like this is easier to read and understand?

File: lib/datasource/adoptium-java/readme.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+Currently only the LTS releases have a JRE available.
```
Comment (HonkingGoose) [inline lib/datasource/adoptium-java/readme.md:8]:
```suggestion
Currently only the LTS releases of Java come with the JRE.
```

I think we mean that only LTS releases of Java come with the Java Runtime Environment?

File: lib/datasource/adoptium-java/readme.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 This datasource returns releases from [Adoptium](https://adoptium.net/) API.
 
-It uses `project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
+It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
 
 It only uses the first 50 pages with 50 items per page.
+
+Use `java-jdk` or `java` as `lookupName` to get releases which come with the Java Development Kit.
+
+Use `java-jre`  as `lookupName` if you only want releases which come with the Java Runtime Environment.
```
Comment (viceice) [inline lib/datasource/adoptium-java/readme.md:9]:
```suggestion
Use `java-jre` as `lookupName` if you only want releases which come with the Java Runtime Environment.
```

**Review by HonkingGoose — APPROVED** [review]
Docs look good now.

**Review by rarkins — APPROVED** [review]

**Pull request merged**